catalog/MDCCatalog/MDCCatalogDebugAlert.swift

```diff
@@ -0,0 +1,200 @@
+
+    layout.minimumLineSpacing = 0
+    layout.minimumInteritemSpacing = 0
+
+    super.init(collectionViewLayout: layout)
+
+    self.transitioningDelegate = dialogTransitionController
+    self.modalPresentationStyle = .custom
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    self.collectionView?.backgroundColor = UIColor.white
+    self.collectionView?.register(MDCCatalogDebugToggleCell.self,
+                                  forCellWithReuseIdentifier: "toggle")
+    self.collectionView?.register(MDCCatalogDebugDismissCell.self,
+                                  forCellWithReuseIdentifier: "dismiss")
+  }
+
+  override func viewWillLayoutSubviews() {
+    super.viewWillLayoutSubviews()
+
+    layout.itemSize = CGSize.init(width: self.view.frame.width, height: 44)
+  }
+
+  override func collectionView(_ collectionView: UICollectionView,
+                               cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+    if indexPath.row == self.dismissCellRow {
+      return collectionView.dequeueReusableCell(withReuseIdentifier: "dismiss", for: indexPath)
+    }
+    let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "toggle", for: indexPath)
+    if let cell = cell as? MDCCatalogDebugToggleCell {
+      cell.setting = settings[indexPath.row]
+    }
+    return cell
+  }
+
+  override func collectionView(_ collectionView: UICollectionView,
+                               numberOfItemsInSection section: Int) -> Int {
+    return self.numberOfRows
+  }
+
+  override func collectionView(_ collectionView: UICollectionView,
+                               didSelectItemAt indexPath: IndexPath) {
+    if indexPath.row == self.dismissCellRow {
+      self.dismiss(animated: true, completion: nil)
+    }
+  }
+}
+
+fileprivate class MDCCatalogDebugToggleCell: UICollectionViewCell {
+  private var _setting: MDCCatalogDebugSetting?
+  var setting: MDCCatalogDebugSetting? {
+    get {
+      return _setting
+    }
+    set {
+      _setting = newValue
+      if let setting = _setting {
+        label.text = setting.title
+        toggleSwitch.isOn = setting.getter()
+      }
+    }
+  }
+
+  let label = UILabel()
+  let toggleSwitch = UISwitch()
+
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    contentView.addSubview(label)
+    contentView.addSubview(toggleSwitch)
+
+    toggleSwitch.addTarget(self, action:#selector(switchToggled(sender:)), for: .touchUpInside)
+  }
+
+  override func layoutSubviews() {
+    super.layoutSubviews()
+
+    let width = self.contentView.bounds.width
+    let height = self.contentView.bounds.height
+
+    let switchSize = toggleSwitch.sizeThatFits(self.contentView.bounds.size)
+    toggleSwitch.frame = CGRect.init(x: width - switchSize.width - 10,
+                                     y: (height - switchSize.height)/2.0,
```

Comment:
I think we do spaces before and after the `/`.

Reply:
Done.